metric_providers/network/connections/proxy/container/provider.py

```diff
@@ -32,7 +32,7 @@ def __init__(self, *, host_ip=None, skip_check=False):
         self._host_ip = host_ip
 
     def check_system(self):
-        super().check_system()
+        self.check_parallel_provider_running()
```

Comment:
you mean super().check_parallel_provider_running() ?

If so, I can - but as far as I know, these two lines are functionally identical. I saw a mixture of both and just standardized it to self.check_parallel_provider() because I didn't really see a difference.

Though thinking about it "out loud" now - super() makes it explicit to the reader that the function being called is in the super class, so stylistically I can see why that'd be preferred. Is there any other reason I'm missing?

Reply:
no, I mean super.check_system()

Why can you not call the complete super function? Why only the check_parallel_provider_running?

Reply:
I think maybe this is an issue in how Github PR's show up - if you saw the whole class it would be clear I think...

This function isn't just calling check_parallel_provider - I am overwriting the check_system with its own logic (in this provider's case, checking the output of the "tinyproxy -v" command. This is different from the base.py's check_system function, which tries to call the binary with -c

The super().check_parallel_provider_running is an additional check on top of that to see if there are multiple instances of this metric provider running. This is shared amongst almost all the providers, which is why its a function from base.by which I call from here (in addition to being part of the default implementation of check_system)